crates/oxc_transformer/src/es2021/logical_assignment_operators.rs

```diff
@@ -122,11 +122,11 @@ impl<'a> Traverse<'a> for LogicalAssignmentOperators<'a> {
         match &mut assignment_expr.left {
             AssignmentTarget::AssignmentTargetIdentifier(ident) => {
                 left_expr = ctx.ast.expression_from_identifier_reference(
-                    Self::clone_identifier_reference(ident, ctx),
+                    ctx.clone_identifier_reference(ident, ReferenceFlag::Read),
                 );
                 assign_target = AssignmentTarget::from(
                     ctx.ast.simple_assignment_target_from_identifier_reference(
-                        Self::clone_identifier_reference(ident, ctx),
+                        ctx.clone_identifier_reference(ident, ReferenceFlag::Write),
                     ),
                 );
```

Comment:
Since you're cloning same `IdentifierReference` twice, it would be preferable to look up the `SymbolId` only once, and then create the 2 x `IdentifierReference`s with `ctx.create_bound_reference_id`.

Reply:
Yes, but that would produce many boilerplate codes.

Reply:
OK. That's fair. I'll add an API to make it less verbose and then we can use that instead.